File: tests/fixtures/java-project/src/main/java/com/example/service/AuthService.java
package com.example.service;

import com.example.data.UserRepo;

public class AuthService {
    private final UserRepo repo;

    public AuthService() {
        this.repo = new UserRepo();
    }

    public boolean login(String username, String password) {
        return repo.findByUsername(username) != null;
    }
}

File: tests/fixtures/csharp-project/Services/AuthService.cs
namespace MyApp.Services;

using MyApp.Data;

public class AuthService
{
    private readonly UserRepository _repo;

    public AuthService(UserRepository repo)
    {
        _repo = repo;
    }

    public bool Login(string username, string password)
    {
        var user = _repo.FindByUsername(username);
        return user != null;
    }
}
